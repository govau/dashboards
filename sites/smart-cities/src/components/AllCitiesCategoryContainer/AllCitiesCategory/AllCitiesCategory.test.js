import React from 'react';
import { shallow } from 'enzyme';
import AllCitiesCategory from './AllCitiesCategory';

jest.mock('../../PageBanner/PageBanner', () => 'PageBanner');
jest.mock('../../SubCategorySummary/SubCategorySummary', () => 'SubCategorySummary');
jest.mock('../../CityColumnChart/CityColumnChart', () => 'CityColumnChart');

const defaultProps = {
  category: {
    colorName: 'jobs',
    description: 'This is the mock category',
    heroIndicatorId: 'population',
    id: 'mock',
    name: 'Mock category',
    subCategories: [
      {
        name: 'Buying houses',
        highlightColorDark: 'JOBS_600',
        summaryIndicatorIds: [],
        charts: [],
      },
      {
        name: 'Living in houses',
        highlightColorDark: 'JOBS_700',
        summaryIndicatorIds: [],
        charts: [
          {
            name: 'Population',
            indicatorIds: ['population'],
          },
        ],
      },
    ],
  },
  cities: [
    {
      name: 'Perth',
      indicators: {}
    },
    {
      name: 'Sydney',
      indicators: {}
    },
  ],
  setShowChartPatterns: () => {},
  showChartPatterns: false,
};

it('should match Snapshot', () => {
  const component = shallow(
    <AllCitiesCategory {...defaultProps} />
  );

  expect(component.debug()).toMatchSnapshot();
});
