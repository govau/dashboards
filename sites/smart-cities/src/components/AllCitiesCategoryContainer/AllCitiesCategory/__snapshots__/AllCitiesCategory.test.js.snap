// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
"<PageWrapper categoryId=\\"mock\\" cityId=\\"all-cities\\">
  <PageBanner colorName=\\"jobs\\" description=\\"This is the mock category\\" indicatorId=\\"population\\" indicatorSubCategory={{...}} title=\\"Mock category\\" cities={{...}} isAllCitiesCategory={true} />
  <SubCategorySummary name=\\"Buying houses\\" highlightColorDark=\\"JOBS_600\\" summaryIndicatorIds={{...}} charts={{...}} categoryColorName=\\"jobs\\" cities={{...}} />
  <SubCategorySummary name=\\"Living in houses\\" highlightColorDark=\\"JOBS_700\\" summaryIndicatorIds={{...}} charts={{...}} categoryColorName=\\"jobs\\" cities={{...}} />
  <SubCategoryCharts subCategory={{...}} colorName=\\"jobs\\" cities={{...}} heroIndicatorId=\\"population\\" highlightColorDark=\\"JOBS_600\\" setShowChartPatterns={[Function]} showChartPatterns={false} />
  <SubCategoryCharts subCategory={{...}} colorName=\\"jobs\\" cities={{...}} heroIndicatorId=\\"population\\" highlightColorDark=\\"JOBS_700\\" setShowChartPatterns={[Function]} showChartPatterns={false} />
</PageWrapper>"
`;
